### ISSUE TITLE
Some License Header Discrepancies are Not Reported in ESH

### ISSUE THREAD

**Issue by dstivanov** [openhab/static-code-analysis#323]
As we can see in [ESH PR #6059](https://github.com/eclipse/smarthome/pull/6059), there were some inconsistencies in the license headers of some java files and these inconsistencies were not reported by the Static Analysis Tooling. We have to check this and fix it.
An initial idea would be to implement simple unit tests with the headers from [ESH PR #6059](https://github.com/eclipse/smarthome/pull/6059) and check why they are missing.

**Comment by lpapazow** [comment]
The reason for these inconsistencies is because these files are not included in the `src/main/java` as our [configuration](https://github.com/openhab/static-code-analysis/blob/master/sat-plugin/src/main/resources/configuration/checkstyle.properties#L3)  requires.

It probably makes sense to apply the **ParameterizedRegexpHeaderCheck** to all files, including those in `src/test/java`, `src/main/feature` and `src/org/eclipse/smarthome/model/item/internal/`.


I can think of 4 ways to do this, each with its pros and flaws

1. Change the configuration to check every file in `src` folder and add a [checkstyle filter](http://checkstyle.sourceforge.net/writingfilters.html) to suppress every warning that does not come from `src/main/java`, unless the warnings created by the **ParameterizedRegexpHeaderCheck**. The problem with this approach is every check for every file will still be executed, which will take time, the warnings will just be suppressed.

2. Change the configuration to check every file in `src` folder and add a custom suppression that looks like this:
```
<suppress files="[\\/]src[\\/]test[\\/].*" checks="^((?!org.openhab.tools.analysis.checkstyle.ParameterizedRegexpHeaderCheck
).)*$" />
```
This will suppress every check in the `src/test` directory that is NOT ParameterizedRegexpHeaderCheck and thus will only execute the ParameterizedRegexpHeaderCheck. The issue with this approach is that the double negative in the suppressing NOT this check is very confusing.


3. Create a custom rules.xml configuration file in every maven submodule, which does not contain `src/main/java` but we still want the ParameterizedRegexpHeaderCheck executed for that module. The problem with this approach is that we need to have many identical files across the smarthome repo.

4. Create another test_rules.xml configuration file in the static-analysis-tool, that only contains ParameterizedRegexpHeaderCheck. Then, for each maven submodule in smarthome, check if the module name is one of previously-defined list of modules that don't have `src/main/java`

@maggu2810 , @kaikreuzer , @martinvw wdyt?
